src/sqlfluff/core/parser/segments/base.py

```diff
@@ -116,14 +116,17 @@ def __init__(
         if not pos_marker:
             # If no pos given, it's the pos of the first segment.
             if isinstance(segments, (tuple, list)):
-                pos_marker = PositionMarker.from_child_markers(
-                    *(seg.pos_marker for seg in segments)
-                )
+                if all(seg.pos_marker for seg in segments):
+                    pos_marker = PositionMarker.from_child_markers(
+                        *(seg.pos_marker for seg in segments)
+                    )
             else:  # pragma: no cover
                 raise TypeError(
                     f"Unexpected type passed to BaseSegment: {type(segments)}"
                 )
-        self.pos_marker: PositionMarker = pos_marker
+
+        # TODO static typing is off for this var throughout the code base
+        self.pos_marker = cast(PositionMarker, pos_marker)
```

Comment:
Now that `BaseSegmentpos_marker` is [marked as `Optional`](https://github.com/sqlfluff/sqlfluff/blob/main/src/sqlfluff/core/parser/segments/base.py#L87), could we remove the `cast`?